### PR TITLE
add function check_guest_machine_type and check_audit_log

### PR DESCRIPTION
**Pull request by nanli1**
need to add function
Signed-off-by: nanli <nanli@redhat.com>

For https://github.com/autotest/tp-libvirt/pull/5657

### CODE REVIEW

File: virttest/utils_libvirt/libvirt_vmxml.py [hunk 1]
```diff
@@ -57,6 +57,26 @@ def check_guest_xml(vm_name, pat_in_dumpxml, option="", status_error=False):
...
+    """
+    Check guest machine type version
+
+    :param vmxml: the guest xml.
+    :param expected_version: expected guest machine version, eg
+    if <type arch='x86_64' machine='pc-q35-rhel8.2.0'>hvm</type> exists,
+    which means expected_version is 8.2.0
+    :return: True or False, the flag of checking successfully or not.
+    """
+    actual_version = int("".join(re.findall(r"\d+", vmxml.os.machine.split("-")[-1])))
```
Comment (dzhengfy) [inline virttest/utils_libvirt/libvirt_vmxml.py:70]: When expected_version="9.10.0" and actual_version= "10.0.0". The function will break.

So perhaps we have to compare the values one by one.

File: virttest/utils_libvirt/libvirt_vmxml.py [hunk 2]
```diff
@@ -57,6 +57,26 @@ def check_guest_xml(vm_name, pat_in_dumpxml, option="", status_error=False):
...
+    expected_version = int(expected_version.replace(".", ""))
+
+    if actual_version < expected_version:
+        LOG.debug("Guest config machine should be >= rhel{}".format(expected_version))
+        return True
```
Comment (dzhengfy) [inline virttest/utils_libvirt/libvirt_vmxml.py:75]: I think we should return False in this condition.

File: virttest/utils_libvirt/libvirt_vmxml.py (hunk 1 above)
Comment (nanli1) [reply virttest/utils_libvirt/libvirt_vmxml.py]: Thanks your reminder, Updated

File: virttest/utils_libvirt/libvirt_vmxml.py (hunk 2 above)
Comment (nanli1) [reply virttest/utils_libvirt/libvirt_vmxml.py]: Thanks , Updated

**Review by dzhengfy — APPROVED** [review]
LGTM

**Pull request merged**